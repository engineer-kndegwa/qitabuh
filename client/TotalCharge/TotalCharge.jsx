import React from 'react';
import PropTypes from 'prop-types';

// Helpers
import { calculateTotalCharge } from "../helpers.js"

const TotalCharge = ({selectedBooks}) => {
    if(!selectedBooks.length) return <div className="no-selected">ðŸ˜¢ No books selected.Please select a title and add a duration to preview cost.</div>;
    const { displayValue } = calculateTotalCharge(selectedBooks);
    return (
        <div className="view-charge">
            ðŸ˜ƒ.Total book charge comes to
        <span>
            {displayValue}
        </span>
        
        </div>
    )
};

TotalCharge.propTypes = {
    selectedBooks: PropTypes.array.isRequired
};

export default TotalCharge;