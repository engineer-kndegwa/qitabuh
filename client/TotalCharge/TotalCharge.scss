.view-charge {
    margin: 2.5em auto;
    text-align: center;
    width: 40%;
    background: $primary-color-2;
    padding: 1em 4em;
    color: $white;
    font-size: 1em;
    border:  8px solid rgba($secondary-color-2, 0.3);
    display: flex;
    justify-content: center;
    align-items: center;
    border-left: none;
    border-right: none;
    border-top: none;
    font-weight: lighter;

    span {
        color: $primary-color;
        margin-left: 1em;
        font-weight: bold;
    }
}