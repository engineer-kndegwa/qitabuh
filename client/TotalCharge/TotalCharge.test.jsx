import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

// Component
import TotalCharge from "./TotalCharge";

const dummyProps = {}

beforeAll(()=>{
    const props = {
        selectedBooks:[]
    };

    const propsAlt = {
        selectedBooks: [{
            id: 'abcd',
            title: "Title",
            durationInDays: 0,
            isSelected: false,
            category: ""
        }]
    };

    dummyProps.props = props;
    dummyProps.propsAlt = propsAlt;
    return dummyProps;
});

test('Should render correctly when all props are passed.', () => {
    const { props, propsAlt } = dummyProps;
    const tree = renderer.create(<TotalCharge {...props} />).toJSON();
    const treeAlt = renderer.create(<TotalCharge {...propsAlt} />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(treeAlt).toMatchSnapshot();
});

test('Should render null when there are no items', () => {
    const { props } = dummyProps;
    const wrapper = shallow(<TotalCharge {...props}/>);
    expect(wrapper.type()).toEqual('div');
    expect(wrapper.containsMatchingElement('<div className="no-selected">ðŸ˜¢ No books selected.Please select a title and add a duration to preview cost.</div>'))
});
