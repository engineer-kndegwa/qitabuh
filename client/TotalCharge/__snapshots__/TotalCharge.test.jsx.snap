// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly when all props are passed. 1`] = `
<div
  className="no-selected"
>
  ðŸ˜¢ No books selected.Please select a title and add a duration to preview cost.
</div>
`;

exports[`Should render correctly when all props are passed. 2`] = `
<div
  className="view-charge"
>
  ðŸ˜ƒ.Total book charge comes to
  <span>
    $0
  </span>
</div>
`;
